src/components/MlEphantConversation2.tsx

```diff
@@ -18,23 +19,89 @@ import type {
 } from '@src/machines/mlEphantManagerMachine2'
 import type { ReactNode } from 'react'
 import { Fragment, useEffect, useRef, useState } from 'react'
+import { DEFAULT_ML_COPILOT_MODE } from '@src/lib/constants'
 
 export interface MlEphantConversationProps {
   isLoading: boolean
   conversation?: Conversation
   contexts: MlEphantManagerPromptContext[]
   billingContext: BillingContext
-  onProcess: (request: string) => void
+  onProcess: (request: string, mode: MlCopilotMode) => void
   disabled?: boolean
   hasPromptCompleted: boolean
   userAvatarSrc?: string
   defaultPrompt?: string
 }
 
+const ML_COPILOT_MODE_META = Object.freeze({
+  fast: {
+    pretty: 'Fast',
+    icon: (props: { className: string }) => (
+      <CustomIcon name="stopwatch" className={props.className} />
+    ),
+  },
+  thoughtful: {
+    pretty: 'Thoughtful',
+    icon: (props: { className: string }) => (
+      <CustomIcon name="brain" className={props.className} />
+    ),
+  },
+} as const)
+
+const ML_COPILOT_MODE: Readonly<MlCopilotMode[]> = Object.freeze([
+  'fast',
+  'thoughtful',
+])
```

Comment:
iirc you only have to use the `as const` trick with Object.freeze with Arrays, not Objects, is that right @lee-at-zoo-corp?